docs/Analyzer Configuration.md

```diff
@@ -455,3 +455,30 @@ Examples:
 |`dotnet_code_quality.CA1710.additional_required_suffixes = MyClass->Class` | All types inheriting from 'MyClass' are expected to have the 'Class' suffix |
 |`dotnet_code_quality.CA1710.additional_required_suffixes = MyClass->Class|MyNamespace.IPath->Path` | All types inheriting from 'MyClass' are expected to have the 'Class' suffix AND all types implementing 'MyNamespace.IPath' are expected to have the 'Path' suffix. |
 |`dotnet_code_quality.CA1710.additional_required_suffixes = T:System.Data.IDataReader->{}` | Allows to override built-in suffixes, in this case, all types implementing 'IDataReader' are no longer expected to end in 'Collection' |
+
+### Inheritance excluded type or namespace names
+Option Name: `additional_inheritance_excluded_symbol_names`
+
+Configurable Rules: [CA1501](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1501)
+
+Option Values: Names of types or namespaces (separated by '|'), such that the type or type's namespace doesn't count in the inheritance hierarchy tree.
+Allowed symbol name formats:
+  1. Type or namespace name (includes all types with the name, regardless of the containing type or namespace and all types whose namespace contains the name)
+  2. Type or namespace name ending with a wildcard symbol (includes all types whose name starts with the given name, regardless of the containing type or namespace and all types whose namespace contains the name)
+  3. Fully qualified names in the symbol's documentation ID format: https://github.com/dotnet/csharplang/blob/master/spec/documentation-comments.md#id-string-format with an optional "T:" prefix for types or "N:" prefix for namespaces.
+  4. Fully qualified wildcard names in the symbol's documentation ID format: https://github.com/dotnet/csharplang/blob/master/spec/documentation-comments.md#id-string-format with an optional "T:" prefix for types or "N:" prefix for namespaces and ending with the wildcard symbol.
```

Comment:
This is not completely true, right? We support all prefixes from the symbol's documentation ID format, but will fail if say someone were to provide `M:MyMethod(System.String)`, which is a valid documentation ID format, but not something we support as we only do a symbol name based search by walking the symbol's containing symbol chain.

Maybe better to not mention documentation ID format in this bullet point, and just state fully qualified name?

Reply:
Sounds good to me but we have the same _incorrect_ behavior for all the recently added options.

Reply:
> we have the same incorrect behavior for all the recently added options

Which ones? I thought none of the previous options were based on wildcards and always used the public APIs on DocumentationCommentId to find the matching symbol for a specified name in documentation ID format.

Reply:
I think I got confused to what was your suggestion. I thought that you wanted me to remove the documentation ID reference for all bullet points because we were accepting only types and namespaces. If this was your suggestion then yes we have added a couple of rules for which we don't support all member kinds allowed by the doc.